examples/opentelemetry-operator/opentelemetry-collector.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: otel-sidecar-collector-sample
+spec:
+  components:
+    - type: "k8s-objects"
+      name: "otel-deploy"
+      properties:
+        objects:
+          - apiVersion: apps/v1
+            kind: Deployment
+            metadata:
+              name: my-app
+              labels:
+                app: my-app
+            spec:
+              selector:
+                matchLabels:
+                  app: my-app
+              replicas: 1
+              template:
+                metadata:
+                  labels:
+                    app: my-app
+                  annotations:
+                    sidecar.opentelemetry.io/inject: "true" # CORRECT
+                spec:
+                  containers:
+                    - name: myapp
+                      image: jaegertracing/vertx-create-span:operator-e2e-tests
+                      ports:
+                        - containerPort: 8080
+                          protocol: TCP
+      traits:
+        - type: opentelemetry-collector
+          properties:
+            mode: sidecar
+            config: |
+              receivers:
+                jaeger:
+                  protocols:
+                    thrift_compact:
+              processors:
+
+              exporters:
+                logging:
+
+              service:
+                pipelines:
+                  traces:
+                    receivers: [jaeger]
```

Comment:
Maybe we also need a `jaeger` addon?

Reply:
Going to file an issue to create `Jaeger` as an addon, then work on it.

Reply:
Done 👍🏽